implementation/src/main/java/io/smallrye/mutiny/groups/UniAndGroup2.java

```diff
@@ -71,4 +71,29 @@ private <O> Uni<O> combine(BiFunction<T1, T2, O> combinator) {
         return super.combinedWith(function);
     }
 
+    /**
+     * Creates the resulting {@link Uni}. The items are combined using the given combinator function,
+     * and the resulting {@code Uni<Uni<O>>} is flattened.
+     *
+     * @param combinator the combinator function, must not be {@code null}
+     * @param <O> the type of item
+     * @return the resulting {@link Uni}. The items are combined into a {@link Tuple2 Tuple2&lt;T1, T2&gt;}.
+     */
+    @CheckReturnValue
+    public <O> Uni<O> combinedWithUni(BiFunction<T1, T2, Uni<O>> combinator) {
+        BiFunction<T1, T2, Uni<O>> actual = Infrastructure.decorate(nonNull(combinator, "combinator"));
+        return combineUni(actual);
+    }
+
+    @SuppressWarnings("unchecked")
+    private <O> Uni<O> combineUni(BiFunction<T1, T2, Uni<O>> combinator) {
+        Function<List<?>, Uni<O>> function = list -> {
+            Tuples.ensureArity(list, 2);
+            T1 item1 = (T1) list.get(0);
+            T2 item2 = (T2) list.get(1);
+            return combinator.apply(item1, item2);
+        };
```

Comment:
extract this fucntion to a constant.

Reply:
This function is not shared. The type of `function` in line 65 is different from `function` in line 90.

Reply:
Yes, but extract it to a constant fun will reduce allocation.

Reply:
But this inner function uses `combinator` from the scope, I can't possibly turn the function into a constant. (But I'd love to be corrected, if you have a comprehensive suggestion)

Also, this pattern of allocating lambdas is all over the repository, don't think this is the right place to start a discussion about how lambda allocation might impact vert.x' performance and memory use.

Reply:
Ouch, I totally missed that!

Reply:
I don't see any performance / allocation issue here

Reply:
As this is a `Uni`, so `UniAndCombination(2....N)` seems not needed anyway.